ocs_ci/ocs/hsbench.py

```diff
@@ -134,36 +146,69 @@ def run_benchmark(
         self.secret_key = secret_key if secret_key else self.secret_key
         self.end_point = end_point if end_point else self.end_point
+        self.num_bucket = num_bucket if num_bucket else self.num_bucket
+        self.object_size = object_size if object_size else self.object_size
         self.pod_obj.exec_cmd_on_pod(
-            f"{self.hsbench_bin_dir} -a {self.access_key} "
+            f"{self.hsbench_bin_dir} "
+            f"-a {self.access_key} "
             f"-s {self.secret_key} "
             f"-u {self.end_point} "
             f"-z {self.object_size} "
-            f"-d {self.duration} -t {self.num_threads} "
+            f"-d {self.duration} "
+            f"-t {self.num_threads} "
             f"-b {self.num_bucket} "
-            f"-n {self.num_obj} -m {self.run_mode} "
-            f"-bp {self.bucket_prefix}",
+            f"-n {self.num_obj} "
+            f"-m {self.run_mode} "
+            f"-bp {self.bucket_prefix} "
+            f"-o {self.output}",
             timeout=timeout,
         )
+        if validate:
+            self.validate_hsbench_workload()
+
+    def validate_hsbench_workload(self):
+        """
+        Validate if workload was running on the app-pod
+
+        Raise:
+            UnexpectedBehaviour: if result.csv file doesn't contain output data.
+        """
+        cmd = (
+            f"cp {self.pod_obj.name}:/go/{self.output} "
+            f"{self.warp_dir}/{self.output}"
+        )
+        self.ocp_obj.exec_oc_cmd(
+            command=cmd,
+            out_yaml_format=False,
+            timeout=180,
+        )
+        if os.path.getsize(f"{self.warp_dir}/{self.output}") != 0:
+            log.info("Workload was running...")
+        else:
+            raise UnexpectedBehaviour(
+                f"Output file {self.output} is empty, "
+                "Warp workload doesn't run as expected..."
+            )
 
     def validate_s3_objects(self):
         """
         Validate S3 objects created by hsbench on single bucket
 
         """
-        self.bucket_name = self.bucket_prefix + "000000000000"
-        num_objects = self.toolbox.exec_sh_cmd_on_pod(
-            f"radosgw-admin bucket stats --bucket={self.bucket_name} | grep num_objects"
-        )
-        if num_objects is not self.num_obj:
-            log.info(
-                f"Number object is created by hsbench in {self.bucket_name} "
-                f"is {num_objects}"
+        for i in range(self.num_bucket):
+            bucket_name = self.bucket_prefix + "00000000000" + str(i)
+            num_objects = self.toolbox.exec_sh_cmd_on_pod(
+                f"radosgw-admin bucket stats --bucket={bucket_name} | grep num_objects"
+            )
+            if num_objects is not self.num_obj:
+                log.info(
+                    f"Number object is created by hsbench in {bucket_name} "
+                    f"is {num_objects}"
+                )
```

Comment:
Would it be possible to also read the content of the objects to check for consistency?

Reply:
hsbench can get objects back and store into result file. Results can be evaluated from the file.

Reply:
Added pre-upgrade, post-upgrade validation including object content comparison.